src/scan/postgres_scan.cpp

```diff
@@ -168,58 +87,47 @@ PostgresReplacementScan(duckdb::ClientContext &context, duckdb::ReplacementScanI
 		return nullptr;
 	}
 
-	/* Check name against query table list and verify that it is heap table */
-	auto relid = FindMatchingRelation(table_name);
+	auto relid = FindMatchingRelation(schema_name, table_name);
 
 	if (relid == InvalidOid) {
 		return nullptr;
 	}
 
-	// Check if the Relation is a VIEW
 	auto tuple = SearchSysCache1(RELOID, ObjectIdGetDatum(relid));
 	if (!HeapTupleIsValid(tuple)) {
 		elog(ERROR, "Cache lookup failed for relation %u", relid);
 	}
 
 	auto relForm = (Form_pg_class)GETSTRUCT(tuple);
 
-	// Check if the relation is a view
-	if (relForm->relkind == RELKIND_VIEW) {
+	if (relForm->relkind != RELKIND_VIEW) {
 		ReleaseSysCache(tuple);
-		return ReplaceView(relid);
+		return nullptr;
 	}
 	ReleaseSysCache(tuple);
 
-	RelOptInfo *node = nullptr;
-	Path *node_path = nullptr;
+	auto oid = ObjectIdGetDatum(relid);
+	Datum viewdef = DirectFunctionCall1(pg_get_viewdef, oid);
+	auto view_definition = text_to_cstring(DatumGetTextP(viewdef));
 
-	if (scan_data->m_query_planner_info) {
-		node = FindMatchingRelEntry(relid, scan_data->m_query_planner_info);
-		if (node) {
-			node_path = get_cheapest_fractional_path(node, 0.0);
-		}
+	if (!view_definition) {
+		elog(ERROR, "Could not retrieve view definition for Relation with relid: %u", relid);
 	}
 
-	/* SELECT query will have nodePath so we can return cardinality estimate of scan */
-	Cardinality nodeCardinality = node_path ? node_path->rows : 1;
-
-	if ((node_path != nullptr && (node_path->pathtype == T_IndexScan || node_path->pathtype == T_IndexOnlyScan))) {
-		auto children = CreateFunctionIndexScanArguments(nodeCardinality, node_path, scan_data->m_query_planner_info,
-		                                                 GetActiveSnapshot());
-		auto table_function = duckdb::make_uniq<duckdb::TableFunctionRef>();
-		table_function->function =
-		    duckdb::make_uniq<duckdb::FunctionExpression>("postgres_index_scan", std::move(children));
-		table_function->alias = table_name;
-		return std::move(table_function);
-	} else {
-		auto children = CreateFunctionSeqScanArguments(nodeCardinality, relid, GetActiveSnapshot());
-		auto table_function = duckdb::make_uniq<duckdb::TableFunctionRef>();
-		table_function->function =
-		    duckdb::make_uniq<duckdb::FunctionExpression>("postgres_seq_scan", std::move(children));
-		table_function->alias = table_name;
-		return std::move(table_function);
+	duckdb::Parser parser;
+	parser.ParseQuery(view_definition);
+	auto statements = std::move(parser.statements);
+	if (statements.size() != 1) {
+		elog(ERROR, "View definition contained more than 1 statement!");
```

Comment:
Let's throw exceptions here instead, given the guidance in `elog(ERROR)` provided in #176